lib/cartopy/img_transform.py

```diff
@@ -358,7 +358,12 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
         if np.ma.isMaskedArray(new_array):
             new_array[non_self_inverse_points] = np.ma.masked
         else:
-            new_array = np.ma.array(new_array, mask=non_self_inverse_points)
+            new_array = np.ma.array(new_array, mask=False)
+            if new_array.ndim == 3:
+                for i in range(new_array.shape[2]):
+                    new_array[non_self_inverse_points, i] = np.ma.masked
```

Comment:
@QuLogic Please don't look at the snippet in isolation. It is pasted from the solution introduced in other locations by https://github.com/SciTools/cartopy/commit/3eb2fd30c587d4a04988e6219b441f7ce11dde2a and overlooked here. Perhaps factoring it out to a local function would be better.
As a side note, this code path only seems to execute when the projection rotates meridians close to horizontal lines. At least this is when my custom projection triggered the exception. This may explain why it went unnoticed for such long time: the standard projections keep meridians verticalish.
@pelson Unfortunately, I don't have a test case. As I have a job now, it is unlikely I will find time to construct it.